internal/api/list_project_events_v1alpha1.go

```diff
@@ -0,0 +1,59 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"connectrpc.com/connect"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/kubeclient"
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+func (s *server) ListProjectEvents(
+	ctx context.Context,
+	req *connect.Request[svcv1alpha1.ListProjectEventsRequest],
+) (*connect.Response[svcv1alpha1.ListProjectEventsResponse], error) {
+	project := req.Msg.GetProject()
+	if err := validateFieldNotEmpty("project", project); err != nil {
+		return nil, err
+	}
+
+	if err := s.validateProjectExists(ctx, project); err != nil {
+		return nil, err
+	}
+
+	var eventsList corev1.EventList
+	if err := s.client.List(
+		ctx,
+		&eventsList,
+		client.InNamespace(req.Msg.GetProject()),
+		// List Kargo related events only
+		client.MatchingFieldsSelector{
+			Selector: fields.OneTermEqualSelector(
+				kubeclient.EventsByInvolvedObjectAPIGroupIndexField,
+				kargoapi.GroupVersion.Group,
```

Comment:
As @rbreeze pointed out API version can be changed later on, so I decided to use `Group` only instead of `GroupVersion`, WDYT?